packages/mui-joy/src/Autocomplete/Autocomplete.tsx

```diff
@@ -668,12 +670,13 @@ const Autocomplete = React.forwardRef(function Autocomplete(
             }
             return renderListOption(option, index);
           })}
-          {loading && groupedOptions.length === 0 ? (
-            <SlotLoading {...loadingProps}>{loadingText}</SlotLoading>
-          ) : null}
           {groupedOptions.length === 0 && !freeSolo && !loading ? (
             <SlotNoOptions {...noOptionsProps}>{noOptionsText}</SlotNoOptions>
           ) : null}
+          {loading &&
+          (showLoadingWithOptions || (!showLoadingWithOptions && groupedOptions.length === 0)) ? (
+            <SlotLoading {...loadingProps}>{loadingText}</SlotLoading>
```

Comment:
i think it is working, you can see `loading` text is displayed in this demo. https://deploy-preview-35653--material-ui.netlify.app/joy-ui/react-autocomplete/#loading

Reply:
Okay, it works. I wonder whether we can by default show loading within the options when `onScrollToBottom` prop is provided to the `Autocomplete`. Then there won't be a need to have a prop `showLoadingWithOptions`.

Reply:
Since current behaviour will break (by break I mean UI will change)I've created new prop but if change is fine then we can remove prop or we keep this prop and remove it in v6

Reply:
Yeah it makes sense,We could do that, shall go ahead with change or wait for approval from team

Reply:
Please go ahead.